### ISSUE TITLE
Failed Resources menu enhancement

### ISSUE THREAD

**Issue by galarosa** [FelisCatus/SwitchyOmega#542]
In the failed resources menu are listed the domains of the failed resources, but there are cases where only a subdomain or specific host is blocked by a particular proxy ( eg. talkgadget.google.com in spite of the entire google.com domain ). Is it possible to report the host of the failed resource giving the possibility to the user to assign e different proxy to the host or the domain by an additional checkbox, may be selected by default?

**Comment by galarosa** [comment]
I've done more tests and some time only the host or the subdomain is reported for the blocked resources but, sometime, as I have stated before the entire domain is reported as blocked when actually it is only an host or a subdomain. It should be useful to know the url of the blocked resources to add a condition manually when the extension does not find the correct rule by itself.

**Comment by FelisCatus** [comment]
For performance reasons, the failed resource count is aggregated by base domain as the requests are being processed, before showing them on the menu.

It is possible to track the exact URL and do whatever processing on the menu, which allows more flexibility. However, I tried that before and the menu took too much time to load. That's why I did that optimization.

**Comment by galarosa** [comment]
Is there a way to know the exact urls processed, may be enabling e debug flag and having the complete urls in the error log?

**Comment by FelisCatus** [comment]
@galarosa If you only want to know the requests, you can use Chrome Developer's Tool by pressing  F12 and switch to the Network tab. All the requests will appear there after you refresh the page and it shows the errors in red.

**Comment by galarosa** [comment]
It's not so simple, when the proxy doesn't give an 4xx or 5xx error for a forbidden resource but gives you a 302 redirecting the browser to a blocking page is difficult in the chrome developer interface to distinguish between 302s from the webserver and 302s from the proxy

**Comment by FelisCatus** [comment]
Then it would be more difficult to tell from SwitchyOmega, which doesn't record the request details and have absolutely no idea about proxy network internals. I would still suggest using Chrome Developer Tools for the availability of almost all information (including headers, etc.) that can be helpful to investigate the issue.

Besides, I want to argue that SwitchyOmega (or any extension) won't do better than Chrome Developer Tools in the case of analyzing requests. Extensions do not have any access to protocol details (no, they cannot talk to proxies over HTTP themselves), and [the webRequest API](https://developer.chrome.com/extensions/webRequest) doesn't expose more information than that in Developer Tools, if not less.

Sorry for not being able to help, as an extension developer.
